### ISSUE TITLE
NullableArray `show`

### ISSUE THREAD

**Issue by quinnj** [JuliaStats/NullableArrays.jl#49]
I think this has been discussed in other places, perhaps more centered on the printing of `Nullable`, but it feels quite messy/unwieldy with the current printing of `NullableArray`. I understand the basic notion that a `NullableArray` represents a vector of `Nullable`, but it makes the actual visual inspection of a `NullableArray` quite a terrible experience. Below is a vector of NullabeVectors (that I'm trying out in CSV.jl)

``` julia
julia> t[1:3]
3-element Array{Any,1}:
 [Nullable("2015-07-01T07:01:50"),Nullable("2015-07-01T07:31:30"),Nullable("2015-07-01T07:05:50"),Nullable("2015-07-01T07:50:00"),Nullable("2015-07-01T07:52:10"),Nullable("2015-07-01T07:02:00"),Nullable("2015-07-01T07:01:10"),Nullable("2015-07-01T07:58:40"),Nullable("2015-07-01T07:31:50"),Nullable("2015-07-01T07:06:20")  …  Nullable("2015-07-01T07:12:29"),Nullable("2015-07-01T07:20:39"),Nullable("2015-07-01T07:24:49"),Nullable("2015-07-01T07:19:19"),Nullable("2015-07-01T07:06:59"),Nullable("2015-07-01T07:17:19"),Nullable("2015-07-01T07:07:39"),Nullable("2015-07-01T07:13:29"),Nullable("2015-07-01T07:35:29"),Nullable{CSV.Str}()]
 [Nullable{CSV.Str}(),Nullable{CSV.Str}(),Nullable("i693f1luPOIr4bXup9rXqg"),Nullable("FRMcvJ2NRt8T-Voq6dfM0Q"),Nullable("txJ3NT1hHUbiHpe9qhmLyA"),Nullable("qqN9baJydqQHk1A1lILpdg"),Nullable("PxkAJ1gk_u6K94q-3Qvd7Q"),Nullable("Ycj7LUDolvG1Z9uHxkAv-A"),Nullable("ErqIs4BK44LaWNPVu6ocbg"),Nullable("IZt5EK626rQy8U7OMYirGw")  …  Nullable("F-jJfCMJ84jtYnCjTHPvmA"),Nullable("5veQetK7eWrn9jkZOBOcYA"),Nullable("FkMA0r11xP5S5aUXDyYj3w"),Nullable("pxvh_7NERcWhurst3dE6WA"),Nullable("GKZu4JC6Q80XRdfOa5urdw"),Nullable("sRZ4F2-g9lRHDt_ll-Nz8g"),Nullable("dJlqWeK8Y42ueNjEgtJySw"),Nullable("Efve2y5pQ8uUxw_hEKkzZA"),Nullable("jXSFg3op-I7G5gLWD2JxPA"),Nullable{CSV.Str}()]
 [Nullable(47174537),Nullable(47174537),Nullable(47174537),Nullable(49899017),Nullable(47174537),Nullable(47776697),Nullable(47174537),Nullable(47174537),Nullable(47174537),Nullable(0)  …  Nullable(47776697),Nullable(47776697),Nullable(47776697),Nullable(47776697),Nullable(47776697),Nullable(47776697),Nullable(47776697),Nullable(47776697),Nullable(47776697),Nullable{Int64}()]
```

Unfortunately, I don't have any great ideas off the top of my head, but perhaps something like:

``` julia
NullableArray{Int64,1}[1, 1, 1, NULL, 0, 0, 0, NULL]
```

Where it's still clear that I have a `NullableArray` and what it's element type is, yet it's much easier to distinguish non-null from null.

**Comment by davidagold** [comment]
I like it. I think you're right to suggest we ought to override `Base.show` at the level of `NullableArray` rather than that of `Nullable`. I'll get started on something.

BTW, if you get a chance to check it out, I'd love to get your thoughts on this `Table` type and related interface mock-up I've been working on: https://github.com/davidagold/Nierika.jl/tree/columns -- the README on that `columns` branch gives a pretty good gist of the current state of the project.

**Comment by quinnj** [comment]
Cool. I'm super slammed at the moment, but I'm hoping to have "caught up" on everything by the end of next week so I can checkout some of the cool stuff you've been working on.

**Comment by davidagold** [comment]
Awesome. I know you're busy, so I appreciate any time you are able to take to look at it. No worries if that can't happen till a later time.

**Comment by quinnj** [comment]
Closed via #50
